### PR TITLE
chore: Update docs and migration guide

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
- Update docs and migration guide
- Bump tracking version
- Fix method usage after merge (method has been removed)
- Fix whitespace
- Fix link
- Improve function and procedure examples (multiline)

References: #3303 #3302 #3298

### CODE REVIEW

File: examples/resources/snowflake_function_scala/resource.tf [hunk 1]
```diff
@@ -31,13 +31,13 @@ resource "snowflake_function_scala" "complete" {
   }
   comment                      = "some comment"
   external_access_integrations = ["external_access_integration_name", "external_access_integration_name_2"]
-  function_definition          = <<EOF
+  function_definition          = <<EOT
```
Comment (sfc-gh-jmichalak) [inline examples/resources/snowflake_function_scala/resource.tf:34]: Note: We should address this later, but we use both EOF and EOT in different files.

File: MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -76,10 +107,16 @@ All of the preview features objects are now disabled by default. This includes:
 	- `snowflake_object_parameter`
 	- `snowflake_password_policy`
 	- `snowflake_pipe`
+    - `snowflake_procedure_java`
+    - `snowflake_procedure_javascript`
+    - `snowflake_procedure_python`
+    - `snowflake_procedure_scala`
+    - `snowflake_procedure_sql`
```
Comment (sfc-gh-jmichalak) [inline MIGRATION_GUIDE.md:114]: Nit: tabs vs spaces :) , below as well.

**Review by sfc-gh-jmichalak — APPROVED** [review]

File: examples/resources/snowflake_function_scala/resource.tf (hunk 1 above)
Comment (sfc-gh-asawicki) [reply examples/resources/snowflake_function_scala/resource.tf:34]: I am pretty sure that now we only have EOT (do you see any files with EOF?)

File: MIGRATION_GUIDE.md (hunk 2 above)
Comment (sfc-gh-asawicki) [reply MIGRATION_GUIDE.md:114]: Changed all to double-space

File: examples/resources/snowflake_function_scala/resource.tf (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply examples/resources/snowflake_function_scala/resource.tf:34]: Masking policy example and acc test.

**Pull request merged**